flight/camera.py

```diff
@@ -87,135 +111,235 @@ async def capture_photo(self, path: str = f"{os.getcwd()}/images/") -> tuple[str
         # So we have to make sure the images folder exists.
         os.makedirs(path, mode=0o777, exist_ok=True)
 
-        file_path = self.camera.capture(gphoto2.GP_CAPTURE_IMAGE)
-        while True:
-            event_type, _event_data = self.camera.wait_for_event(100)
-            if event_type == gphoto2.GP_EVENT_CAPTURE_COMPLETE:
+        photo_result = self.camera.requestPhoto()
```

Comment:
This needs typing but honestly I am not too worried about it since this is the only two lines it is used